src/java/org/apache/cassandra/service/ActiveRepairService.java

```diff
@@ -155,6 +157,8 @@
      * A map of active coordinator session.
      */
     private final ConcurrentMap<UUID, RepairSession> sessions = new ConcurrentHashMap<>();
+    public final ConcurrentMap<UUID, CoordinatorState> repairs = new ConcurrentHashMap<>();
+    public final ConcurrentMap<UUID, ValidationState> validations = new ConcurrentHashMap<>();
```

Comment:
should we clear them after repair is completed? Or we can move them into `ParentRepairSession` which should have the same lifecycle as `CoordinatorState` and `ValidationState`

Reply:
I need to add a cleanup logic to keep this all in-sync.  IR uses a scheduled task, should do the same with a configurable expiration window (based off completion time); will add

Reply:
based off feedback in the streaming vtable patch, moved to `Cache`.  The cache is size based, and the size is based off the estimated of the payload

Reply:
> should we clear them after repair is completed? 
No, one big point of this feature is the ability to look at a repair while its running *and* after the fact; many times we know something went wrong and we want to see what the repair did and that is scattered in the logs or just not present; dropping after the repair is complete means we have to go back to looking at the logs.

> Or we can move them into ParentRepairSession

that gets cleaned up when the cleanup message is sent, it would be nice to keep the state around longer.